WorkflowFunctionAdminBundle/Security/Authorization/Voter/WorkflowStateVoter.php

```diff
@@ -0,0 +1,142 @@
+use FOS\UserBundle\Model\UserInterface;
+
+/**
+ * Class WorkflowStateVoter
+ */
+class WorkflowStateVoter implements VoterInterface
+{
+    protected $workflowRightRepository;
+    protected $contentTypeRepository;
+
+    /**
+     * @param WorkflowRightRepositoryInterface $workflowRightRepository
+     * @param ContentTypeRepositoryInterface   $contentTypeRepository
+     */
+    public function __construct(WorkflowRightRepositoryInterface $workflowRightRepository, ContentTypeRepositoryInterface $contentTypeRepository)
+    {
+        $this->workflowRightRepository = $workflowRightRepository;
+        $this->contentTypeRepository = $contentTypeRepository;
+    }
+
+    /**
+     * Checks if the voter supports the given attribute.
+     *
+     * @param string $attribute An attribute
+     *
+     * @return bool true if this Voter supports the attribute, false otherwise
+     */
+    public function supportsAttribute($attribute)
+    {
+        return 0 === strpos($attribute, 'ROLE_ACCESS_UPDATE_NODE') ||
+               0 === strpos($attribute, 'ROLE_ACCESS_UPDATE_ERROR_NODE') ||
+               0 === strpos($attribute, 'ROLE_ACCESS_UPDATE_CONTENT_TYPE_FOR_CONTENT');
+    }
+
+    /**
+     * Checks if the voter supports the given class.
+     *
+     * @param string $class A class name
+     *
+     * @return bool true if this Voter can process the class
+     */
+    public function supportsClass($class)
+    {
+        return is_subclass_of($class, 'OpenOrchestra\ModelInterface\Model\StatusableInterface');
+    }
+
+    /**
+     * Returns the vote for the given parameters.
+     *
+     * This method must return one of the following constants:
+     * ACCESS_GRANTED, ACCESS_DENIED, or ACCESS_ABSTAIN.
+     *
+     * @param TokenInterface $token      A TokenInterface instance
+     * @param object|null    $object     The object to secure
+     * @param array          $attributes An array of attributes associated with the method being invoked
+     *
+     * @return int either ACCESS_GRANTED, ACCESS_ABSTAIN, or ACCESS_DENIED
+     */
+    public function vote(TokenInterface $token, $object, array $attributes)
+    {
+        $user = $token->getUser();
+        $isOrchestraUser = $user instanceof UserInterface;
+        if ($isOrchestraUser && $user->isSuperAdmin()) {
+            return self::ACCESS_GRANTED;
+        }
+        if (!$this->supportsClass($object)) {
+            return self::ACCESS_ABSTAIN;
+        }
+        foreach ($attributes as $role) {
+            if (!$this->supportsAttribute($role)) {
+                return self::ACCESS_ABSTAIN;
+            }
+        }
+        if ($isOrchestraUser) {
+            $isOwner = $object instanceof BlameableInterface && ($object->getCreatedBy() == $user->getUsername());
+            $workflowRight = $this->workflowRightRepository->findOneByUserId($user->getId());
+            if (null === $workflowRight) {
+                return VoterInterface::ACCESS_DENIED;
```

Comment:
@itkg-ngilain  This part is out of the scope, i create a ticket for speak of this problem